Add Scale utils.py

sarahbx opened the pull request:
Add `tests/scale/utils.py`

Part of a dependency path for scale upgrade testing #3031
Rebased on #3103 

Depends on:
#3029 - Move Chaos utils
#3103 - Move EUS Constant


##### Short description:

##### More details:

##### What this PR does / why we need it:

##### Which issue(s) this PR fixes:

##### Special notes for reviewer:

##### jira-ticket:
<!--  full-ticket-url needs to be provided. This would add a link to the pull request to the jira and close it when the pull request is merged
If the task is not tracked by a Jira ticket, just write "NONE".
-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Refactor**
  * Consolidated VM nginx service provisioning and HTTP reachability checks into shared utilities for reuse.

* **Tests**
  * Added a fixture to run multiprocessing with the fork start method and applied it across chaos tests.
  * Added new scale test helpers to manage machine config pool scenarios and measure operation timings.
  * Updated chaos tests to use the consolidated reachability utilities.

* **Chores**
  * Added a shared EUS constant and made MCP wait timing parameters configurable.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

sarahbx committed:
Move chaos utils

Move chaos utils for use in other modules
- create_vm_with_nginx_service
- verify_vm_service_reachable
- is_http_ok

sarahbx committed:
Add typing, rename args

sarahbx committed:
Move EUS constant to fix import error

sarahbx committed:
Add scale utils.py

sarahbx committed:
Add typing, make name required

rnetser commented on tests/scale/utils.py:
```diff
@@ -0,0 +1,63 @@
+from ocp_resources.resource import ResourceEditor
+
+
+def get_user_kubeconfig_context(kubeconfig_filename: str, username: str) -> str:  # skip-unused-code
```
why is this function needed? can you please provide some context via a doscrting?

sarahbx replied:
done

sarahbx committed:
Alter chaos multiprocessing start method

The previous fix of using get_context(method) is no longer working.
To remedy this, switching to a fixture that sets the
multiprocessing start method instead.
https://github.com/python/cpython/issues/132898
https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Process

sarahbx committed:
Add docstring to get_user_kubeconfig_context

sarahbx committed:
Add MachineConfigPoolConfiguration ExitStack

sarahbx committed:
Update unittests for operator.py

sarahbx committed:
Add additional unittetst assertion

openshift-virtualization-qe-bot-3 commented:
/retest all

_Auto-triggered: Files in this PR were modified by merged PR #3002._

<details>
<summary>Overlapping files</summary>

tests/chaos/utils.py

</details>
